internal/services/mssql/mssql_job_step_resource_test.go

```diff
@@ -92,6 +92,45 @@ func TestAccMsSqlJobStep_complete(t *testing.T) {
 	})
 }
 
+func TestAccMsSqlJobStep_withManagedIdentity(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mssql_job_step", "test")
+	r := MsSqlJobStepTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.withManagedIdentity(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("job_credential_id").IsEmpty(),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccMsSqlJobStep_updateFromCredentialToManagedIdentity(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mssql_job_step", "test")
+	r := MsSqlJobStepTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.withCredential(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("job_credential_id").IsSet(),
+			),
+		},
+		{
+			Config: r.withManagedIdentity(data),
```

Comment:
Once the test configs are updated we can replace `withManagedIdentity` with `basic`

```suggestion
			Config: r.basic(data),
```